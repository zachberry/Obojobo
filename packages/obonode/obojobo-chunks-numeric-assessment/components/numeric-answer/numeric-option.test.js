import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import NumericOption from './numeric-option'

describe('NumericOption', () => {
	test('NumericOption renders as expected with requirement of `exact`', () => {
		const component = renderer.create(<NumericOption numericChoice={{ requirement: 'exact' }} />)

		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericOption with requirement of `exact`, event.stopPropagation() is called when click', () => {
		const component = mount(<NumericOption numericChoice={{ requirement: 'exact' }} />)

		const stopPropagation = jest.fn()

		component
			.find('.select-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		const tree = component.html()

		expect(tree).toMatchSnapshot()
		expect(stopPropagation).toBeCalledTimes(2)
	})

	test('NumericOption renders as expected with requirement of `range`', () => {
		const component = renderer.create(<NumericOption numericChoice={{ requirement: 'range' }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericOption with requirement of `range`, event.stopPropagation() is called when click', () => {
		const component = mount(<NumericOption numericChoice={{ requirement: 'range' }} />)

		const stopPropagation = jest.fn()
		component
			.find('.select-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})

		const tree = component.html()

		expect(tree).toMatchSnapshot()
		expect(stopPropagation).toBeCalledTimes(3)
	})

	test('NumericOption renders as expected with requirement of `margin`', () => {
		const component = renderer.create(<NumericOption numericChoice={{ requirement: 'margin' }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericOption with requirement of `margin`, event.stopPropagation() is called when click', () => {
		const component = mount(<NumericOption numericChoice={{ requirement: 'margin' }} />)

		const stopPropagation = jest.fn()

		component
			.find('.select-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.select-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})
		const tree = component.html()

		expect(tree).toMatchSnapshot()
		expect(stopPropagation).toBeCalledTimes(4)
	})

	test('NumericOption renders as expected with requirement of `precise`', () => {
		const component = renderer.create(<NumericOption numericChoice={{ requirement: 'precise' }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericOption with requirement of `precise`, event.stopPropagation() is called when click', () => {
		const component = mount(<NumericOption numericChoice={{ requirement: 'precise' }} />)

		const stopPropagation = jest.fn()
		component
			.find('.select-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.select-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})

		const tree = component.html()

		expect(tree).toMatchSnapshot()
		expect(stopPropagation).toBeCalledTimes(4)
	})

	test('NumericOption renders as expected with default requirement', () => {
		const component = renderer.create(<NumericOption numericChoice={{ requirement: '' }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
