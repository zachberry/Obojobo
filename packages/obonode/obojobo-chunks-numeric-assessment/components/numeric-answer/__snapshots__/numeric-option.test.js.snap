// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericOption NumericOption renders as expected with default requirement 1`] = `
<tr>
  <td>
    <select
      className="select-item"
      name="requirement"
      onClick={[Function]}
    >
      <option>
        Exact answer
      </option>
      <option>
        Margin of error
      </option>
      <option>
        Within a range
      </option>
      <option>
        Precise response
      </option>
    </select>
  </td>
  <td>
    <input
      className="input-item"
      contentEditable={false}
      name="answer"
      onClick={[Function]}
      value=""
    />
  </td>
</tr>
`;

exports[`NumericOption NumericOption renders as expected with requirement of \`exact\` 1`] = `
<tr>
  <td>
    <select
      className="select-item"
      name="requirement"
      onClick={[Function]}
      value="Exact answer"
    >
      <option>
        Exact answer
      </option>
      <option>
        Margin of error
      </option>
      <option>
        Within a range
      </option>
      <option>
        Precise response
      </option>
    </select>
  </td>
  <td>
    <input
      className="input-item"
      contentEditable={false}
      name="answer"
      onClick={[Function]}
      value=""
    />
  </td>
</tr>
`;

exports[`NumericOption NumericOption renders as expected with requirement of \`margin\` 1`] = `
<tr>
  <td>
    <select
      className="select-item"
      name="requirement"
      onClick={[Function]}
      value="Margin of error"
    >
      <option>
        Exact answer
      </option>
      <option>
        Margin of error
      </option>
      <option>
        Within a range
      </option>
      <option>
        Precise response
      </option>
    </select>
  </td>
  <td>
    <select
      className="select-item"
      name="type"
      onClick={[Function]}
    >
      <option>
        Percent
      </option>
      <option>
        Absolute
      </option>
    </select>
  </td>
  <td>
    <input
      className="input-item"
      name="answer"
      onClick={[Function]}
      value=""
    />
  </td>
  <td>
    <input
      className="input-item"
      name="margin"
      onClick={[Function]}
      value=""
    />
  </td>
</tr>
`;

exports[`NumericOption NumericOption renders as expected with requirement of \`precise\` 1`] = `
<tr>
  <td>
    <select
      className="select-item"
      name="requirement"
      onClick={[Function]}
      value="Precise response"
    >
      <option>
        Exact answer
      </option>
      <option>
        Margin of error
      </option>
      <option>
        Within a range
      </option>
      <option>
        Precise response
      </option>
    </select>
  </td>
  <td>
    <select
      className="select-item"
      name="type"
      onClick={[Function]}
    >
      <option>
        Significant digits
      </option>
      <option>
        Decimal places
      </option>
    </select>
  </td>
  <td>
    <input
      className="input-item"
      name="answer"
      onClick={[Function]}
      value=""
    />
  </td>
  <td>
    <input
      className="input-item"
      name="precision"
      onClick={[Function]}
      value=""
    />
  </td>
</tr>
`;

exports[`NumericOption NumericOption renders as expected with requirement of \`range\` 1`] = `
<tr>
  <td>
    <select
      className="select-item"
      name="requirement"
      onClick={[Function]}
      value="Within a range"
    >
      <option>
        Exact answer
      </option>
      <option>
        Margin of error
      </option>
      <option>
        Within a range
      </option>
      <option>
        Precise response
      </option>
    </select>
  </td>
  <td>
    <input
      className="input-item"
      name="start"
      onClick={[Function]}
      value=""
    />
  </td>
  <td>
    <input
      className="input-item"
      name="end"
      onClick={[Function]}
      value=""
    />
  </td>
</tr>
`;

exports[`NumericOption NumericOption with requirement of \`exact\`, event.stopPropagation() is called when click 1`] = `"<tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><input class=\\"input-item\\" name=\\"answer\\" contenteditable=\\"false\\" value=\\"\\"></td></tr>"`;

exports[`NumericOption NumericOption with requirement of \`margin\`, event.stopPropagation() is called when click 1`] = `"<tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><select class=\\"select-item\\" name=\\"type\\"><option>Percent</option><option>Absolute</option></select></td><td><input class=\\"input-item\\" name=\\"answer\\" value=\\"\\"></td><td><input class=\\"input-item\\" name=\\"margin\\" value=\\"\\"></td></tr>"`;

exports[`NumericOption NumericOption with requirement of \`precise\`, event.stopPropagation() is called when click 1`] = `"<tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><select class=\\"select-item\\" name=\\"type\\"><option>Significant digits</option><option>Decimal places</option></select></td><td><input class=\\"input-item\\" name=\\"answer\\" value=\\"\\"></td><td><input class=\\"input-item\\" name=\\"precision\\" value=\\"\\"></td></tr>"`;

exports[`NumericOption NumericOption with requirement of \`range\`, event.stopPropagation() is called when click 1`] = `"<tr><td><select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option><option>Precise response</option></select></td><td><input class=\\"input-item\\" name=\\"start\\" value=\\"\\"></td><td><input class=\\"input-item\\" name=\\"end\\" value=\\"\\"></td></tr>"`;
